app/routes/_auth+/onboarding.tsx

```diff
@@ -81,9 +82,25 @@ export async function action({ request }: DataFunctionArgs) {
 	}
 
 	const formData = await request.formData()
-	const submission = parse(formData, {
-		schema: onboardingFormSchema,
+	const submission = await parse(formData, {
+		schema: () => {
+			return onboardingFormSchema.superRefine(async (data, ctx) => {
```

Comment:
Could this be:

```suggestion
		schema: onboardingFormSchema.superRefine(async (data, ctx) => {
```

I'm not sure of the benefit of using a callback here.

Reply:
@edmundhung is there a reason using the callback is better?

Reply:
It is needed only when you are setting up async live validation. This is likely a leftover from the previous implementation.

Reply:
> async live validation

What does this mean?

Reply:
Async validation on input or blur event

Reply:
I don't understand why that would need the callback form on the server.

Reply:
It's an advanced feature. The callback is used to retrieve the `submission.intent` and adjust the zod schema behavior accordingly. In the case of async validation on input / blur, this allows us to run async validation only when it is intent to.

You can find an example [here](https://conform.guide/validation#skipping-validation)

Reply:
Oh I see! Thank you for explaining that. I'm really amazed of all the things you've thought of with this!